torch/distributed/_tensor/device_mesh.py

```diff
@@ -269,3 +272,51 @@ def get_coordinate(self) -> Optional[List[int]]:
         dimensions of the mesh. If this rank is not part of the mesh, return None.
         """
         return self._coordinate_on_dim if self._coordinate_on_dim else None
+
+
+def init_device_mesh(
+    device_type: str,
+    mesh_shape: Tuple[int],
+    *,
+    mesh_dim_names: Optional[Tuple[str]] = None,
+) -> DeviceMesh:
+    """
+    Initializes a `DeviceMesh` based on `device_type`, `mesh_shape`, and `mesh_dim_names` parameters.
+    This creates a DeviceMesh with a mesh layout of n-d dimensional array, n being the len(mesh_shape)
+    and ith dimension being in size mesh_shape[i]. If mesh_dim_names is provided, each dimension is
+    labeled as mesh_dim_names[i].
+
+
+    Args:
+        device_type (str): device type of the mesh. Currently supports: cpu, cuda/cuda-like.
+        mesh_shape: Tuple[int]: A tuple describes the dimension of the multi-dimesnion array
+        that describes the layout of devices.
+    Kwargs:
+        mesh_dim_names: Optional[Tuple[str]]: A tuple of mesh dim names to be assigned to each dimension
+        of the multi-dimensional array that describes the layout of devices. Its length must match the length
+        of `mesh_shape`.
+
+    Returns:
+        A :class:`DeviceMesh` object
+
+    .. note: If no process group is found, init_device_mesh will initialize distributed process group/groups
+    behind the scene, which are requried for distributed communications.
+
+    Example:
+        >>> # xdoctest: +SKIP
+        >>> two_d_mesh = init_device_mesh("cuda", mesh_shape=(2, 8), mesh_dim_names=("dp", "tp"))
+        >>> two_d_mesh = init_device_mesh("cuda", mesh_shape=(2, -1), mesh_dim_names=("dp", "tp"))
+    """
+    if mesh_dim_names is not None and len(mesh_shape) != len(mesh_dim_names):
+        raise RuntimeError(
+            f"Please provide a mesh_dim_name to each mesh_dim! Found {len(mesh_dim_names)} instead of {len(mesh_shape)}."
+        )
+
+    mesh = torch.arange(dist.get_world_size()).view(mesh_shape)
```

Comment:
Ye. Thanks for catching the mistake! 

My thought is since we are planning to use init_device_mesh() as an alternative for init_process_group, maybe let's just remove this syntax sugar -1 option for init_device_mesh(). I think use should know the world_size and the mesh_shape(without -1).

If folks really want this option, we can add it back.

Reply:
Yeah that make sense to me!